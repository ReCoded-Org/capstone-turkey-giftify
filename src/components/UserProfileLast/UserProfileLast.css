.userProfileDetailsParentProfilePicDiv > img {
  width: 30%;
  margin-top: 3%;
  border-radius: 15px;
}

svg {
  font-size: 25px;
}

.userProfileParentDiv {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: center;
  align-content: stretch;
}

.userProfileDetailsMainDiv {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: stretch;
  align-content: stretch;
  background-color: #90e0ef;
  width: 50%;
  border-radius: 15px;
  margin: 3% 0%;
}

.userProfileDetailsParentProfilePicDiv {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;
  margin: 3% 0%;
}

.userProfileDetailsParentDiv {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-items: center;
  align-content: stretch;
  margin: 1.5% 0%;
}

.userProfileDetailsParentDiv > p {
  min-width: 35%;
  text-align: center;
  background-color: #f8f9fa;
  border-radius: 15px;
  padding: 1.5% 3%;
  font-size: 1.15em;
}

.userProfileDetailsChildDiv {
  width: 35%;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: flex-start;
  align-content: stretch;
}

.userProfileDetailsChildDiv {
  font-size: 1.25em;
}

@media only screen and (min-width: 638px) and (max-width: 955px) {
  .userProfileDetailsMainDiv {
    width: 75%;
  }
}

@media only screen and (min-width: 120px) and (max-width: 637px) {
  .userProfileDetailsMainDiv {
    width: 85%;
  }

  .userProfileDetailsChildDiv {
    font-size: 1.1em;
    font-weight: bold;
  }

  svg {
    font-size: 20px;
    margin: 0% 0% 3% 0%;
    cursor: pointer;
  }

  .userProfileDetailsChildDiv {
    display: flex;
    flex-direction: column;
    flex-wrap: nowrap;
    justify-content: center;
    align-items: center;
    align-content: stretch;
    width: 75%;
  }

  .userProfileDetailsChildDiv {
    margin: 12% 0% 0% 0%;
  }

  .userProfileDetailsParentDiv {
    margin: 0;
  }
}
