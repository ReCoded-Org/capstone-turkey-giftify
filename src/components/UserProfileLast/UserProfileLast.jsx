import React, { useEffect, useState } from "react";
import "./UserProfileLast.css";
import db from "./../../firebase";
import { Gear } from "react-bootstrap-icons";

const UserProfileLast = () => {
  // eslint-disable-next-line
  const [userId, SetUserId] = useState("fIoIImalBCOZt9O7AcgrFMQrLOH3");
  const [userDetails, setUserDetails] = useState([]);

  const getUserDetails = async () => {
    await db
      .collection("users")
      .doc(userId)
      .get()
      .then((doc) => {
        setUserDetails((prevState) => {
          return {
            ...prevState,
            [doc.id]: [
              doc.data().city,
              doc.data().firstName,
              doc.data().lastName,
              doc.data().photoURL,
              doc.data().email,
            ],
          };
        });
      });
  };

  const editUserName = async () => {
    const editedUserName = prompt("User's name: ");
    await db
      .collection("users")
      .doc(userId)
      .update({
        firstName: editedUserName,
      })
      .then(getUserDetails());
  };

  const editUserLastname = async () => {
    const editedUserLastame = prompt("User's lastname: ");
    await db
      .collection("users")
      .doc(userId)
      .update({
        lastName: editedUserLastame,
      })
      .then(getUserDetails());
  };

  const editUserCity = async () => {
    const editedUserCity = prompt("User's city: ");
    await db
      .collection("users")
      .doc(userId)
      .update({
        city: editedUserCity,
      })
      .then(getUserDetails());
  };

  const editUserMail = async () => {
    const editedUserMail = prompt("User's mail: ");
    await db
      .collection("users")
      .doc(userId)
      .update({
        email: editedUserMail,
      })
      .then(getUserDetails());
  };

  const editUserProfilePic = async () => {
    const editedUserPic = prompt("User's profile pic: ");
    await db
      .collection("users")
      .doc(userId)
      .update({
        photoURL: editedUserPic,
      })
      .then(getUserDetails());
  };

  useEffect(() => {
    getUserDetails();
    // eslint-disable-next-line
  }, []);

  return (
    <div className="userProfileParentDiv">
      {Object.values(userDetails).map((info) => (
        <div className="userProfileDetailsMainDiv" key={info[4]}>
          <div className="userProfileDetailsParentProfilePicDiv">
            <Gear onClick={editUserProfilePic} />
            <img src={info[3]} alt={info[1]} />
          </div>

          <div className="userProfileDetailsParentDiv">
            <div className="userProfileDetailsChildDiv">
              <p>User's Name</p>
              <Gear onClick={editUserName} />
            </div>
            <p>{info[1]}</p>
          </div>

          <div className="userProfileDetailsParentDiv">
            <div className="userProfileDetailsChildDiv">
              <p>User's Lastname</p>
              <Gear onClick={editUserLastname} />
            </div>
            <p>{info[2]}</p>
          </div>

          <div className="userProfileDetailsParentDiv">
            <div className="userProfileDetailsChildDiv">
              <p>User's City</p>
              <Gear onClick={editUserCity} />
            </div>
            <p>{info[0]}</p>
          </div>

          <div className="userProfileDetailsParentDiv">
            <div className="userProfileDetailsChildDiv">
              <p>User's Mail</p>
              <Gear onClick={editUserMail} />
            </div>
            <p>{info[4]}</p>
          </div>
        </div>
      ))}
    </div>
  );
};

export default UserProfileLast;
