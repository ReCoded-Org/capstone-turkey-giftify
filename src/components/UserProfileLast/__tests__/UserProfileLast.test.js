import React from "react";
import UserProfileLast from "./../UserProfileLast";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("UserProfileLast rendered correctly", () => {
  const tree = renderer.create(<UserProfileLast />).toJSON();
  expect(tree).toMatchSnapshot();
});
