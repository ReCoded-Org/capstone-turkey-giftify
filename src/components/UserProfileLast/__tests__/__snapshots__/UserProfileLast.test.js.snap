// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileLast rendered correctly 1`] = `
<div
  className="userProfileParentDiv"
/>
`;
